### ISSUE TITLE
Unable to store result of unionall in let query variable

### ISSUE THREAD

**Issue by creisle** [orientechnologies/orientdb#8955]
### Java Version
openjdk version "1.8.0_191"
OpenJDK Runtime Environment (build 1.8.0_191-b12)
OpenJDK 64-Bit Server VM (build 25.191-b12, mixed mode)
## Expected behavior  
Using the demodb I expected the following queries to both return the same result

```sql
select expand($d) 
  let $a = (select * from Countries where Name like 'A%'), 
       $b = (select * from Countries where Name like 'B%'), 
       $c = UNIONALL($a, $b), 
       $d = (select * from (select expand($c)) where Name like '%a');
```

```sql
select expand($d) 
  let $a = (select * from Countries where Name like 'A%'), 
       $b = (select * from Countries where Name like 'B%'), 
       $d = (select * from (select expand(UNIONALL($a, $b))) where Name like '%a');
```

## Actual behavior  
The first query returns nothing, The second query returns the expected 18 records

## Steps to reproduce  

Run the queries outlined above

**Comment by creisle** [comment]
Note that this used to work in v2

**Comment by luigidellaquila** [comment]
Hi @creisle 

I have a fix for this, I'm pushing it to 3.0.x and develop branches now

Thanks

Luigi
